### DISCUSSION TITLE
Purging deleted photos

### DISCUSSION THREAD

**Discussion by Gnopps** [photoprism/photoprism#1317] [Q&A] [answered]
Thanks for developing this app!

I've got about 26k photos & videos indexed with some duplicates. I ran a dupecheck and removed several photos & videos from my _Originals_-folder. I then wanted to clean the PhotoPrism database of the dupes I removed.

So I went to Library and started an index operation with "Complete Rescan" unchecked. The result was that about 600 items were moved to _Hidden_. I tried the same operation again but this time with "Complete Rescan" checked. The new result was that _Hidden_ now had about 400 items instead of 600. 

Wanting to get rid of all _Hidden_ items I ran `purge` from the CLI. Result was that 16 files and 484 photos were removed. I then tried running  `purge --hard` but this time nothing was removed although several items with "...has no jpeg" were reported in the output.

So on to the questions:
1. How come purge command removed photos after the index operations had been run? I thought the purge command was run automatically by PhotoPrism after indexing.
2. Any ideas what might cause the difference in _Hidden_ files depending on if I did a complete rescan or not? I had not done any changes to files, database except for the dupe removal before I did my first index operation.
3. What is the proper way to get rid of the last items in _Hidden_? Archive->Delete  doesn't work as they are hidden so can it be done automatically?

Thanks in advance!

**Comment by lastzero** [comment]
It's possible that there are files without a JPEG, like sidecar files or RAWs. Did you check that? Just because you deleted a JPEG doesn't mean there are no related files left.

**Comment by lastzero** [comment, marked as answer]
Also if they were duplicates, it's possible the same file was discovered somewhere else or stacking changed due to different file names. Without knowing what files with what directory structure, metadata and settings you deleted specifically, it's impossible to give you an answer on those counts. To get a more deterministic behaviour, you may reduce the number of indexing workers to 1 so that only one file is indexed at the same time.
